web_autoflip_demo/autoflip_demo_UI/autoflip_worker.ts

```diff
@@ -175,94 +239,131 @@ onmessage = function (e: MessageEvent): void {
         )
         .then((buffer): void => {
           autoflipModule.setAspectRatio(videoAspectWidth, videoAspectHeight);
+          autoflipModule.changeBinaryGraph(buffer);
 
-          // Sets the aspect ratio setAspectRatio(1, 1) = '1:1' or
+          // Sets the aspect ratio setAspectRatio(1, 1) = '1:1'
           // setAspectRatio(16, 11) = '16: 11'.
-          autoflipModule.changeBinaryGraph(buffer);
+          // autoflipModule.setAspectRatio(2, 1);
+          //autoflipModule.cycleGraph();
         });
     });
   }
-  // Gets frameData from indexDB and process with Autoflip.
-  readFramesFromIndexedDB(signal.videoId, signal.startTime).then(
-    (value: Frame[]): void => {
-      console.log(`PROMISE: promise ${signal.videoId} returned`);
-      let frameData: Frame[] = value;
-      handleFrames(frameData, signal);
-      if (signal.end === true) {
-        const b = autoflipModule.closeGraphInternal();
-        // This posts the analysis result back to main script.
-        ctx.postMessage({
-          type: 'finishedAnalysis',
-          cropWindows: resultCropInfo,
-          startId: currentId,
-          videoId: signal.videoId,
-          shots: resultShots,
-        });
-      } else {
-        ctx.postMessage({
-          type: 'nextCrop',
-          videoId: signal.videoId + 1,
-        });
-      }
-    },
-  );
+  if (signal.type === 'nextCropMiss') {
+    console.log(`wait 2 sec!`);
+    setTimeout(() => {
+      console.log(`finish 2 sec!`);
+      ctx.postMessage({
+        type: 'nextCrop',
+        videoId: signal.videoId,
+        startId: signal.startId,
+        user: signal.user,
+      });
+    }, 2000);
+  } else {
+    console.log(`frameNumber middle`, frameNumber);
+    if (signal.type === 'changeAspectRatio') {
+      frameNumber = (1 + signal.videoId) * 15 * workerWindow - signal.startId;
+    } else {
+      frameNumber = workerWindow * 15;
+    }
+    // Gets frameData from indexDB and process with Autoflip.
+    readFramesFromIndexedDB(signal.videoId, signal.startId, frameNumber).then(
+      (value: Frame[]): void => {
+        console.log(`PROMISE: promise ${signal.videoId} returned`);
+        let frameData: Frame[] = value;
+        handleFrames(frameData, signal);
+        if (signal.end === true) {
+          const b = autoflipModule.closeGraphInternal();
+          // This posts the analysis result back to main script.
+          ctx.postMessage({
+            type: 'finishedAnalysis',
+            cropWindows: resultCropInfo,
+            startId: signal.startId,
+            videoId: signal.videoId,
+            shots: resultShots,
```

Comment:
set the features to an empty array like you do below

Reply:
Fixed.